engine/access/rpc/backend/backend_scripts.go

```diff
@@ -205,3 +216,49 @@ func (b *backendScripts) tryExecuteScript(ctx context.Context, executorAddress s
 	}
 	return execResp.GetValue(), nil
 }
+
+func (b *backendScripts) tryExecuteScriptOnArchiveNode(ctx context.Context, executorAddress string, blockID flow.Identifier,
+	script []byte,
+	arguments [][]byte) ([]byte, error) {
+	req := &access.ExecuteScriptAtBlockIDRequest{
+		BlockId:   blockID[:],
+		Script:    script,
+		Arguments: arguments,
+	}
+	port, err := findPortFromAddress(executorAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	archiveClient, closer, err := b.connFactory.GetAccessAPIClientWithPort(executorAddress, port)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to create client for archive node %s: %v", executorAddress, err)
+	}
+	defer closer.Close()
+	resp, err := archiveClient.ExecuteScriptAtBlockID(ctx, req)
+	if err != nil {
+		if status.Code(err) == codes.Unavailable {
+			b.connFactory.InvalidateAccessAPIClient(executorAddress)
+		}
+		return nil, status.Errorf(status.Code(err), "failed to execute the script on archive node %s: %v", executorAddress, err)
+	}
+	return resp.GetValue(), nil
+}
+
+func findPortFromAddress(address string) (uint, error) {
+	_, portStr, err := net.SplitHostPort(address)
+	if err != nil {
+		return 0, fmt.Errorf("fail to extract port from archive address %v: %w", address, err)
```

Comment:
```suggestion
		return 0, fmt.Errorf("fail to extract port from address %v: %w", address, err)
```

function name does not mention anything about archive, so the error should be more general.